src/Adapter/Preferences/PreferencesConfiguration.php

```diff
@@ -28,30 +28,41 @@
 
 use Cookie;
 use PrestaShop\PrestaShop\Adapter\Configuration;
-use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Adapter\Shop\Context;
+use PrestaShop\PrestaShop\Core\Configuration\AbstractMultistoreConfiguration;
+use PrestaShop\PrestaShop\Core\Feature\FeatureInterface;
 use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagSettings;
 use PrestaShopBundle\Entity\Repository\FeatureFlagRepository;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * This class will provide Shop Preferences configuration.
  */
-class PreferencesConfiguration implements DataConfigurationInterface
+class PreferencesConfiguration extends AbstractMultistoreConfiguration
 {
     /**
      * @var Configuration
      */
-    private $configuration;
+    protected $configuration;
 
     /**
      * @var FeatureFlagRepository
      */
     protected $featureFlagRepository;
 
-    public function __construct(
-        Configuration $configuration,
-        FeatureFlagRepository $featureFlagRepository
-    ) {
+    /**
+     * AbstractMultistoreConfiguration constructor.
+     *
+     * @param Configuration $configuration
+     * @param FeatureFlagRepository $featureFlagRepository
+     * @param Context $shopContext
+     * @param FeatureInterface $multistoreFeature
+     */
+    public function __construct(Configuration $configuration, FeatureFlagRepository $featureFlagRepository, Context $shopContext, FeatureInterface $multistoreFeature)
+    {
         $this->configuration = $configuration;
+        $this->shopContext = $shopContext;
+        $this->multistoreFeature = $multistoreFeature;
         $this->featureFlagRepository = $featureFlagRepository;
     }
```

Comment:
As new attributes aren't optional and we said that services constructor changes aren't BC, I recommend to change constructor attributes order to keep some consistency with abstract architecture.
I would recommend to call parent constructor and to keep it's constructor signature order.

```suggestion
    public function __construct(Configuration $configuration, Context $shopContext, FeatureInterface $multistoreFeature, FeatureFlagRepository $featureFlagRepository)
    {
        parent::__construct($configuration, $shopContext, $multistoreFeature)
        $this->featureFlagRepository = $featureFlagRepository;
    }
```


> :information_source:  Not mandatory for me.